### PR TITLE
`docs`: remove `.` from check and format commands

### PR DESCRIPTION
**Pull request by hoel-bagard**
## Summary

This PR modifies the documentation to use `ruff check` instead of `ruff check .`, and `ruff format` instead of `ruff format .`, as discussed [here](https://github.com/astral-sh/ruff/pull/10168#discussion_r1509976904)

### CODE REVIEW

File: docs/formatter.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ The Ruff formatter is available as of Ruff [v0.1.2](https://astral.sh/blog/the-r
 directories, and formats all discovered Python files:
 
 ```shell
-ruff format .                 # Format all files in the current directory.
+ruff format                 # Format all files in the current directory.
```
Comment (MichaReiser) [inline docs/formatter.md:14]:
```suggestion
ruff format                  # Format all files in the current directory.
```

**Review by MichaReiser — COMMENTED** [review]
I'm conflicted on this. While `.` is not strictly necessary, it shows users how to pass a specific `path`. Removing the path everywhere might indicate that ruff doesn't accept custom paths.

**Comment by hoel-bagard** [comment]
@MichaReiser 

> I'm conflicted on this. While . is not strictly necessary, it shows users how to pass a specific path. Removing the path everywhere might indicate that ruff doesn't accept custom paths.

That was also part of my reasoning when I left it in the preview mode documentation PR. I made this PR to have consistent documentation.

> Maybe we leave it in one example and mention that the path can be omitted?

Looking at the documentation, the `.` is used a lot, but doesn't seem to be explicitly explained that much. There is an explanation [here](https://docs.astral.sh/ruff/configuration/#python-file-discovery), and "examples" [here](https://docs.astral.sh/ruff/linter/#detecting-unused-suppression-comments), but one might expect a short explanation on   how to run ruff on a file/directory to be easier to find.
The note in the [ruff check section](https://docs.astral.sh/ruff/linter/#ruff-check) does point to an explanation with examples, but maybe the note could slightly expand on the fact that `ruff check` takes a path as input ?

**Comment by charliermarsh** [comment]
I think I'd personally prefer to keep these, since in my view at least it's more explicit and clearer for readers that don't know that Ruff defaults to `.`.

**Comment by hoel-bagard** [comment]
@zanieb Should I close this PR ?

**Comment by zanieb** [comment]
Humph... I want to get rid of these because people think they're needed and they're really not. It degrades the idea that we're a workspace oriented tool. Checking a specific path seems like a less common behavior that we could document separately.

**Review by zanieb — APPROVED** [review]
Sorry for the back-and-forth, I pushed a couple additional touch-ups and attempted to address Micha's concern.

**Pull request merged**